### PR TITLE
chore: use templated noxfile.py

### LINKED ISSUE #365: Noxfile is not updated with Sphinx version pin
`noxfile.py` is not updated with Sphinx version pin, please help take a look!

### PR DESCRIPTION
**Pull request by parthea**
Fixes #365 🦕

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -185,7 +194,7 @@ def cover(session):
     test runs (not system test runs), and then erases coverage data.
     """
     session.install("coverage", "pytest-cov")
-    session.run("coverage", "report", "--show-missing", "--fail-under=99")
+    session.run("coverage", "report", "--show-missing", "--fail-under=100")
```
Comment (larkee) [inline noxfile.py:197]: This PR is currently failing due to coverage. Due to generated tests not having full coverage, we cannot reach 100% coverage which is why we had `--fail-under=99`.

Comment (larkee) [reply noxfile.py:197]: `docs-presubmit` is currently failing across other PRs so I'd like to get this PR merged but merging is blocked by this line. I'm going to revert this change for now and we can address it in a future PR :+1: .

```suggestion
    session.run("coverage", "report", "--show-missing", "--fail-under=99")
```

Comment (larkee) [reply noxfile.py:197]: Ah. The bot does not like that. I see. Hmmmm :thinking:

Comment (parthea) [reply noxfile.py]: Fixed

**Comment by parthea** [comment]
@ larkee Please could you take another look?

**Comment by parthea** [comment]
@larkee Please could you take another look?

**Comment by parthea** [comment]
@larkee friendly ping

**Review by larkee — APPROVED** [review]

**Pull request merged**